packages/model-viewer/src/test/features/scene-graph/material-spec.ts

```diff
@@ -235,4 +238,45 @@ suite('scene-graph/material', () => {
       expect(element.model!.materials[2].getAlphaMode()).to.be.equal('MASK');
     });
   });
+  suite('Material lazy loading', () => {
+    let element: ModelViewerElement;
+    let model: Model;
+    setup(async () => {
+      element = new ModelViewerElement();
+    });
+
+    teardown(() => {
+      document.body.removeChild(element);
+    });
+
+    const loadModel = async (path: string) => {
+      element.src = path;
+      document.body.insertBefore(element, document.body.firstChild);
+      await waitForEvent(element, 'load');
+      model = element.model as Model;
+    };
+
+    test('Accessing the name getter does not cause throw error.', async () => {
+      await loadModel(CUBES_GLTF_PATH);
+      // tslint:disable-next-line:no-unused-expression
+      expect(() => {model.materials[2].name}).does.not.throw;
+      expect(model.materials[2][$lazyLoadGLTFInfo]).to.be.ok;
+    });
+
+    test(
+        'Accessing a getter of an unloaded material throws an error.',
+        async () => {
+          await loadModel(CUBES_GLTF_PATH);
+          expect(() => {model.materials[2].pbrMetallicRoughness}).to.throw;
```

Comment:
Maybe also check that `isValid` is false?

Reply:
Done